src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private int capacity;
+    private final double loadFactory;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap(int capacity, double loadFactory) {
+        this.capacity = capacity;
+        this.loadFactory = loadFactory;
+        this.table = new Node[capacity];
+    }
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY, LOAD_FACTOR);
+    }
 
     @Override
-    public void put(K key, V value) {
+    public int getSize() {
+        return this.size;
+    }
 
+    @Override
+    public void put(K key, V value) {
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        Node<K, V> newNode = new Node<>(key, value);
+        newNode.next = table[index];
+        table[index] = newNode;
+        size++;
+        if (size > capacity * loadFactory) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
-    public int getSize() {
-        return 0;
+    public void remove(K key) {
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        Node<K, V> prev = null;
+        while (node != null) {
+            if (node.key.equals(key)) {
+                if (prev == null) {
+                    table[index] = node.next;
+                } else {
+                    prev.next = node.next;
+                }
+                size--;
+                return;
+            }
+            prev = node;
+            node = node.next;
+        }
+    }
+
+    @Override
+    public void resize() {
+        int newCapacity = table.length * 2;
+        capacity = (int) (DEFAULT_CAPACITY * newCapacity);
```

Comment:
Here is an example
we had default map with capacity 16 (and table length also 16 obviously)
then we have first resize
newCapacity = 16 * 2 = 32
capacity = 16 * 32 = 512. WHY???